CondFormats/CTPPSReadoutObjects/plugins/TotemDAQMappingESSourceXML.cc

```diff
@@ -419,9 +431,14 @@ void TotemDAQMappingESSourceXML::ParseTreeRP(ParseType pType, xercesc::DOMNode *
     // store mask data
     if (pType == pMask && type == nChip)
     {
+      unsigned int armIdx = (parentID / 1000) % 10; 
+      unsigned int stIdx = (parentID / 100) % 10; 
+      unsigned int rpIdx = (parentID / 10) % 10; 
+      unsigned int plIdx = parentID % 10; 
```

Comment:
Please don't duplicate code. The four values on lines 406-409, 417-420, and 434-437 should be defined once, maybe at line 401. They should also be `const`.